docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,752 @@
+    "d_loss_fun = nn.BCELoss()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Evaluation of custom gradients for the generator BCE loss function"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "outputs": [],
+   "source": [
+    "def g_loss_fun_grad(qnn: QuantumCircuit,\n",
+    "                    parameters: Union[ParameterVector, ParameterExpression, List[ParameterExpression]],\n",
+    "                    param_values: Iterable,\n",
+    "                    discriminator_: nn.Module,\n",
+    "                    grad_method: str = 'param_shift') -> Iterable:\n",
+    "    \"\"\"\n",
+    "    Custom gradient of the generator loss function considering the custom gradients of the quantum generator\n",
+    "    Args:\n",
+    "        qnn: Quantum neural network ansatz given as a quantum circuit.\n",
+    "        parameters: The parameters of the quantum neural network which are trained.\n",
+    "        param_values: The current values of the quantum neural network parameters.\n",
+    "        discriminator_: Classical neural network representing the discriminator.\n",
+    "        grad_method: Method used to compute the gradients {'param_shift', 'lin_comb', 'fin_diff'}\n",
+    "    Returns:\n",
+    "        List of gradient values, i.e., the gradients of the loss function w.r.t. the quantum neural network parameters\n",
+    "    \"\"\"\n",
+    "    grads = generator_grad(qnn, parameters, param_values, grad_method = grad_method)\n",
+    "    loss_grad = ()\n",
+    "    for j, grad in enumerate(grads):\n",
+    "        cx = grad[0].tocoo()\n",
+    "        input = []\n",
+    "        target = []\n",
+    "        weight = []\n",
+    "        for index, prob_grad in zip(cx.col, cx.data):\n",
+    "            input.append(grid_elements[index])\n",
+    "            target.append([1.])\n",
+    "            weight.append([prob_grad])\n",
+    "        bce_loss_grad = F.binary_cross_entropy(discriminator_(Tensor(input)), Tensor(target), weight=Tensor(weight))\n",
+    "        loss_grad += (bce_loss_grad, )\n",
+    "    loss_grad = stack(loss_grad)\n",
+    "    return loss_grad"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Definition of the optimizers"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Initialize generator and discriminator\n",
+    "generator = generator_(qnn, ansatz.ordered_parameters)\n",
```

Comment:
as stated before, we could just instantiate the `CircuitQNN` here instead of using this `generator_()` method

Reply:
just curious, is there any specific reason why the parameters are sent via `ansatz.ordered_parameters` instead of `ansatz.parameters`??